validator/sawtooth_validator/journal/completer.py

```diff
@@ -27,38 +27,51 @@
 from sawtooth_validator.networking.dispatch import Handler
 from sawtooth_validator.networking.dispatch import HandlerResult
 from sawtooth_validator.networking.dispatch import HandlerStatus
+from sawtooth_validator.journal.block_wrapper import NULL_BLOCK_IDENTIFIER
 
 LOGGER = logging.getLogger(__name__)
 
 
 class Completer(object):
-    def __init__(self, block_store):
+    def __init__(self, block_store, gossip):
+        self.gossip = gossip
         self.batch_cache = TimedCache()
         self.block_cache = BlockCache(block_store)
+        # avoid throwing away the genesis block
+        self.block_cache[NULL_BLOCK_IDENTIFIER] = None
         self._on_block_received = None
         self._on_batch_received = None
 
     def _check_block(self, block):
```

Comment:
Since this method now has actions it takes, _check_block() probably isn't a great name for it.